lib/utils/circularityUtils.ts

```diff
@@ -0,0 +1,172 @@
+import {
+  ContentItemElementsIndexer,
+  Elements,
+  IContentItem,
+  IContentItemElements,
+  IContentItemsContainer,
+  IPagination,
+} from "@kontent-ai/delivery-sdk";
+
+/**
+ * Module responsible for detecting and removing circular references
+ * in content items from the Kontent.ai Delivery SDK.
+ */
+
+export type CircularReferenceInfo = {
+  parentItemCodename: string;
+  elementName: string;
+  cycle: string[];
+};
+
+export type ItemCircularReferenceMap = Record<string, CircularReferenceInfo[]>;
+
+type ElementWithLinkedItems =
+  | Elements.LinkedItemsElement
+  | Elements.RichTextElement;
+
+interface IContentItemListingResponse<T extends IContentItem> {
+  items: T[];
+  pagination: IPagination;
+  linkedItems: IContentItemsContainer;
+}
+
+/**
+ * Identifies whether a given element can contain linked items.
+ * @param obj - The element to check.
+ * @returns True for rich text and linked items elements, false otherwise.
+ */
+const isElementWithLinkedItems = (
+  obj: ContentItemElementsIndexer
+): obj is ElementWithLinkedItems => {
+  return (
+    "linkedItems" in obj &&
+    Array.isArray((obj as ElementWithLinkedItems).linkedItems)
+  );
+};
+
+/**
+ * Performs a depth-first traversal to sanitize any content item references that would create a cycle.
+ * Collects information on identified circular references.
+ * @param data - The content item to sanitize.
+ * @param elementName - The name of the current element being processed.
+ * @param seenCodenames - Set of content item codenames visited during the recursion.
+ * @param circularReferences - Accumulated list of detected circular references.
+ * @returns A tuple containing the sanitized content item and the accumulated circular references.
+ */
+const sanitizeCircularDataInternal = <TItem extends IContentItem>(
+  data: TItem,
```

Comment:
why is this called `data` and not `item`?

Reply:
oversight when renaming :) fixed